test/data/config.json

```diff
@@ -77,607 +77,607 @@
 		"secret": [
 			{
 				"encryptedSecret":
-					"19e9f8eee5d9516234a10953669518bf23371d34e114713c8043a98378fd866834946380c4cc0e40f23305643206c1c5be496074f350d09d87735c42eae30c604cabea9862f4fe8f906313c67a7146d54fa844171e6cf925b7c953987082cdd6",
+					"$5$rounds=1$e33b33c7621fc33d361ff26d1b9611b4$aedfce0b845f7c962ea57a76f7bb7e48228273356fdab39ce71c9c385e105f7660101a7ffbee14ef5aa9dc8d03d6715e0828acae8d8625bbec9a167839d0f935445d14a874e18cbed8b01f7da74d990e$iv=51b98dbc815996de7821e56fcc985fbb$tag=75a9571b9018e59255f5ed8ac8f7db75$version=1.0.0",
```

Comment:
Why we used `$` as the key separator? Means we can use any `|` or `-`, is there any specific reason to use `$` symbol? Agree or not `$` is in general used for a special meaning for variables in web development in javascript. So we should rather use some other symbol as the key separator for our hashed/encrypted strings.

Reply:
Yes, it conforms to the existing standard taken from Linux shadow files.

Reply:
@willclarktech I don't feel its a standard. Correct me if I am wrong. The convention was introduced in 1987, and ported to Linux in 1992. Later ported to different linux distros with the variety of changes. The modern linux distro like `Ubuntu` don't seems to follow that convention of `$` symbol. 

Here is some excerpt from Ubuntu shadow file. 

```
root:*:16176:0:99999:7:::
daemon:*:17596:0:99999:7:::
bin:*:17596:0:99999:7:::
sys:*:17596:0:99999:7:::
sync:*:17596:0:99999:7:::
```

It seems to use `:` as key separator. Correct me if I am missing something.

Reply:
@nazarhussain Do you have an alternative proposal?

Reply:
@webmaster128 Thanks for the info. `$pbkdf2-sha256` is in any case more explicit than `$5`.

Reply:
For the record, the idea as discussed with @fchavant @Nazgolze @4miners and @SargeKhan, was not to adopt this as a rigid standard, but to take it as a starting point. For one thing we have hex strings here, whereas the original standard seems to assume base64.

Reply:
Having a custom format is great because in this case there is no format widely adopted by the industry. If you don't mind, let me think out loud. I suggest:

* Having a version is a good thing
* Decouple the format version from the Lisk core version. The format may stay stable between Core 1.0.0 and 3.8 and suddenly change for both branches 3.9 and 4.0. What's the version then?
* Use an integer version. When using 1.0.0 or 2.3.4 you need to parse and handle all the possible versions in between. But there is no value in having multiple version positions. Either bump from version 1 to version 2, or don't.
* Hardcode the parameters `aes-256-gcm`, `pdkdf2-sha256` for version `1`. If you need to change them, bump the version
* Use binary concatenation for the encrypted data: `iv | ciphertext | tag`. `iv` and `tag` have a fixed length, so they can be cut out easily. This has the advantage that you only need 1 field for the entire ciphertext of aes-256-gcm and you do not have algorithm-specific parameters
* [Use a 96bit IV for aes-256-gcm](https://crypto.stackexchange.com/questions/41601/aes-gcm-recommended-iv-size-why-12-bytes)

Then only the fields `version`, `rounds`, `salt`, `ciphertext` are left, and can be encoded using whatever separator (maybe not `$`) or just json: `'{"version": 1, "rounds": 1, "salt": "aabbccdd", "ciphertext": "5c0dc4e86063b4a3814c7f08d9759a98b99ff7d2541eee5cd49ea290dd22dabfee18d23775c5a9f47f7ecbb50b15953c2abc"}'`

Reply:
@willclarktech I checked your message late. I second @webmaster128 for having `custom format`.